### PR TITLE
chore(expo sdk): upgrade to 48

### PR DESCRIPTION
**Pull request by ardasnturk**
This PR upgrades Expo SDK to 48. After deleting yarn.lock and node_modules to reinstall everything from scratch the problem was solved. So maybe you also need to do these steps.

please make a new build to try the new version:
`yarn prebuild`
`yarn ios -d` or `yarn android -d`

Expo [CHANGELOG](https://github.com/expo/expo/blob/main/CHANGELOG.md#4800--2023-02-09)

⚠️ orientation bug fixed in this version ⚠️

SVA-937

### CODE REVIEW

File: app.json [hunk 1]
```diff
@@ -9,7 +9,6 @@
     "platforms": ["ios", "android"],
     "version": "3.1.2",
     "runtimeVersion": "nativeVersion",
-    "orientation": "portrait",
```
Comment (donni106) [inline app.json:12]: why was this removed?

https://docs.expo.dev/versions/latest/config/app/#orientation

File: App.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ const sentryApi = secrets[namespace].sentryApi;
 if (sentryApi?.dsn) {
   Sentry.init({
     dsn: sentryApi.dsn,
+    enableInExpoDevelopment: true,
```
Comment (donni106) [inline App.js:15]: what is the reason for the decision to report dev errors to sentry?

File: app.json [hunk 3]
```diff
@@ -67,6 +67,12 @@
           "microphonePermission": false
         }
       ],
+      [
+        "expo-screen-orientation",
+        {
+          "initialOrientation": "DEFAULT"
+        }
```
Comment (donni106) [inline app.json:74]: does this make our orientation provider superfluous?

File: app.json (hunk 1 above)
Comment (ardasnturk) [reply app.json:12]: since the `initialOrientation` value is set to `DEFAULT` in the plugin section, we do not need the `orientation` value here. if we set the `orientation` value to `portrait`, we cannot switch the application to landscape mode in our builds for iOS.

File: App.js (hunk 2 above)
Comment (donni106) [reply App.js:15]: please check the line two lines after this one, where a comment describes the situation

File: package.json
Comment (donni106) [inline package.json:1]: did you read about `expo-image`? do we want to give it a try?

https://blog.expo.dev/expo-sdk-48-ccb8302e231#d1dc
https://docs.expo.dev/versions/v48.0.0/sdk/image

Comment (ardasnturk) [reply package.json:1]: of course we can try. But I'm not sure it works perfectly. 🤷‍♂️ there is a video example in the tweet I sent below. but I don't know which version of `expo-image` is used in the video.
https://twitter.com/vadimnotjustdev/status/1636631047868653571?s=46&t=TWDbqoteDDeGe0cb3GWRlQ

File: app.json (hunk 3 above)
Comment (donni106) [reply app.json:73]: good info.
1. buttons should not be full width on large screens. that is too large.
2. according full width of contents we decided, that we do not want to smallen anymore, right?!
so we still need the provider but maybe need some adjustments?
but this should go in a separate ticket and process.

File: App.js (hunk 2 above)
Comment (ardasnturk) [reply App.js]: I did rebase and deleted `enableInExpoDevelopment`.

https://github.com/smart-village-solutions/smart-village-app-app/commit/2c91de40ca84dc76fb38888ce7606499e6ea5f5e

File: app.json (hunk 3 above)
Comment (donni106) [reply app.json:73]: i don't have in mind if it's just about the button or maybe anything else. the carousel image is also something i can think of.

File: app.json (hunk 1 above)
Comment (ardasnturk) [reply app.json:12]: yes you are right the lock has been removed in `OnboardingManager`. but why don't we want the application to be rotated at startup?

Comment (donni106) [reply app.json:12]: i think it was related to splash screen images being bad at landscape mode. maybe you find some git history entry about it from the past?

Comment (ardasnturk) [reply app.json:12]: I have tried different builds and now I think that we don't need to set `initialOrientation` to `DEFAULT`.
It is enough to add `expo-screen-orientation` as a plugin.
Initially the application is locked in portrait mode, but afterwards it can be switched to landscape mode when locked to `DEFAULT` mode.

File: app.json (hunk 3 above)
Comment (donni106) [reply app.json]: can you please check if there is anything else orientation aware like the button? the carousel or html image on top of screens?

File: package.json
Comment (donni106) [reply package.json:1]: can you take some time to try it out and spot some differences?

File: app.json (hunk 1 above)
Comment (donni106) [reply app.json]: i built new development apps locally for android and ios.
on android, the orientation works fine.
sadly i have a lot of the following warnings, which i did not knew from earlier versions:
image

on ios, i cannot change orientation in the dev app. when i do an upside down, i see that the status bar goes away, but the screen stays.

any idea, why?

Comment (ardasnturk) [reply app.json]: I don't know the cause of the image source null error. This is the first time I've seen it.

---
I'm sorry to hear that. It worked fine for me. I'll try it and let you know more.

File: app.json (hunk 3 above)
Comment (donni106) [reply app.json]: and what about androids input change on landscape? is this normal?

|portrait|landscape|
|---|---|
|Screenshot_1681733850|Screenshot_1681733863|

do we and the users want/expect it the same on landscape as on portrait?

File: app.json (hunk 1 above)
Comment (ardasnturk) [reply app.json]: After my recent tests, I think this is the code that makes Orientation work the most correctly.

image

The application initially opens in Portrait mode and can then be switched to Landscape mode.

|landscape|portrait|
|--|--|
Simulator Screen Shot - iPhone 14 Plus - 2023-04-19 at 13 04 57 | Simulator Screen Shot - iPhone 14 Plus - 2023-04-19 at 13 04 59

File: app.json (hunk 3 above)
Comment (ardasnturk) [reply app.json]: In this version we use `WrapperWithOrientation` component. In version `3.2.1` this component will be removed. We use `WrapperWithOrientation` component in many places but `noFlex` property is used only by `TextBlock`. We use `needLandscapeStyle` for Landscape. This is used by `Button`, `Checkbox` and `OrientationAwareIcon`.

---
I think it is caused by the `react-native-elements` update. If you try it in version `3.1.3`, it looks the same.
I have encountered Input in this way in other Android applications.

File: package.json
Comment (ardasnturk) [reply package.json:1]: Yes, I will try and write about my experiences. 👍

Comment (ardasnturk) [reply package.json:1]: I must say I'm very impressed with the `expo-image` package. I think the props are very useful and future-proof.
The image component of the `rn-elements` we currently use usually contains style properties. The `expo-image` package has functional features in addition to style features. It's hard to predict which props we'll use and which we won't at the moment, but I can tell you that it will make things a lot easier for a feature we want to do in the future.

For example, I think the `enableLiveTextInteraction` prop is very useful. The `expo-image` package also has its own cache system. I think we can use this as well.

I have not encountered a performance problem like the twitter link I sent before. It may be because I have not done a very big experiment. I am not sure.

I think we can use it.

File: app.json (hunk 3 above)
Comment (donni106) [reply app.json]: Is there a prop to avoid that?

File: package.json
Comment (donni106) [reply package.json:1]: > For example, I think the enableLiveTextInteraction prop is very useful. The expo-image package also has its own cache system. I think we can use this as well.

Why do you think these are useful?

File: app.json (hunk 3 above)
Comment (ardasnturk) [reply app.json]: I think this little code will solve our problem. 👍
image

File: package.json
Comment (ardasnturk) [reply package.json:1]: I think we will be able to use this feature in the future for any requests from customers. (`enableLiveTextInteraction`)

Comment (donni106) [reply package.json:1]: ok i don't fully understand, but let's keep that in mind

File: app.json (hunk 3 above)
Comment (donni106) [reply app.json]: and how does it look like now with the change? what is the effect?

Comment (ardasnturk) [reply app.json]: Looks the same as before.

**Review by donni106 — APPROVED** [review]
